src/main/java/httpserver/HTTPSocket.java

```diff
@@ -0,0 +1,34 @@
+package httpserver;
+
+import java.io.*;
+import java.net.Socket;
+
+public class HTTPSocket implements ClientSocket {
+
+    private final Socket socket;
+
+    public HTTPSocket(Socket socket) {
+        this.socket = socket;
+    }
+
+    @Override
+    public void close() {
+        try {
+            socket.close();
+        } catch (IOException e) {
+            throw new SocketConnectionException("Cannot close socket: ", e.getCause());
+        }
+    }
+
+    @Override
+    public void sendResponse(HTTPResponse httpResponse) {
+        try {
+            OutputStream outputStream = socket.getOutputStream();
+            PrintWriter printWriter = new PrintWriter(outputStream);
+            printWriter.write(httpResponse.HTTP() + " " + httpResponse.getStatusCode() + " " + httpResponse.getReasonPhrase());
+            printWriter.close();
```

Comment:
Does all those methods have a good reason to be into a try?

If
``` Java
PrintWriter printWriter = new PrintWriter(outputStream); 
printWriter.write(httpResponse.HTTP() + " " + httpResponse.getStatusCode() + " " + httpResponse.getReasonPhrase());
```
Can possibly throw exception you may want to put *printWriter.close()* into a finally